acai-ios/ViewControllers/PaymentViewController.swift

```diff
@@ -0,0 +1,207 @@
+            .font: UIFont.avenirNextMedium.withSize(24),
+            .foregroundColor: UIColor.black
+        ]
+    }
+
+    @objc func backButtonPressed() {
+        navigationController?.popViewController(animated: true)
+    }
+
+    func resetTextFields() {
+        cardNumber = ""
+        cardHolder = ""
+        expirationDate = ""
+        cvc = ""
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+    }
+
+}
+
+extension PaymentViewController: UITableViewDelegate, UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        // +1 for label text
+        return inputItems.count + 1
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        if indexPath.row == 0 {
+            let cell = tableView.dequeueReusableCell(withIdentifier: FileConstants.labelCellReuseIdentifier, for: indexPath) as! TextTableViewCell
+            cell.configure(for: "Your information will always remain private and never shared.")
+            return cell
+        } else {
+            let cell = tableView.dequeueReusableCell(withIdentifier: FileConstants.inputCellReuseIdentifier, for: indexPath) as! InputTableViewCell
+            let inputItem = inputItems[indexPath.row - 1]
+            cell.changeInputTextFieldDelegate = self
+            cell.invalidEntryLabel.isHidden = true
+            if let type = inputItem.type {
+                if type == .cardNumber {
+                    cell.textField.text = cardNumber
+                    if attemptedTransaction {
+                        cell.invalidEntryLabel.isHidden = cardNumber.isValidCardNumber()
+                    }
+                } else if type == .cardHolder {
+                    cell.textField.text = cardHolder
+                    if attemptedTransaction {
+                        cell.invalidEntryLabel.isHidden = cardHolder.isValidCardHolder()
+                    }
+                } else if type == .expirationDate {
+                    cell.textField.text = expirationDate
+                    if attemptedTransaction {
+                        cell.invalidEntryLabel.isHidden = expirationDate.isValidExpirationDate()
+                    }
+                } else if type == .cvc {
+                    cell.textField.text = cvc
+                    if attemptedTransaction {
+                        cell.invalidEntryLabel.isHidden = cvc.isValidCVC()
+                    }
+                }
+            }
+            cell.configure(for: inputItem)
+            return cell
+        }
+    }
+
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return FileConstants.inputViewHeight
+    }
+}
+
+extension PaymentViewController: ChangeInputTextFieldDelegate {
+    func changeText(text: String, type: InputType) {
+        if type == .cardNumber {
```

Comment:
if we are matching on enum, prefer to use `switch` instead

Reply:
doesn't list out all the cases